### PR TITLE
Upgrade `ONNX` to 1.13, `protobuf` to 3.20.3 and relax `tensorflow`

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - The packages cannot be bumped in separate PRs because `onnx==1.12` is incompatible with `protobuf==3.20.2`
 - ONNX 1.13 requires `protobuf>=3.20.2`
 - ONNX 1.13 requires C++17 on Windows https://github.com/onnx/onnx/pull/4470
 - Allow `tensorflow==1.12`, which allows higher `protobuf` version
 - Remove deprecated ONNX utility `onnx.mapping.NP_TYPE_TO_TENSOR_TYPE`
 - Skip paddlepaddle tests until a new version is released, with compliant protobuf version

### Tickets:
 - CVS-99694, CVS-96733

### CODE REVIEW

File: src/bindings/python/tests/test_onnx/test_backend.py
Comment (rkazants) [inline src/bindings/python/tests/test_onnx/test_backend.py:67]: why do we have so many xfail tests? We reduce code-coverage this way

**Review by rkazants — CHANGES_REQUESTED** [review]
why do we have so many xfail tests? We reduce code-coverage this way

**Comment by p-wysocki** [comment]
@rkazants Each ONNX release includes new/changes to operators and functions, which come along with new ONNX backend tests. Those are independent from us and show up only when upgrading the package. 

The xfails here can be divided into two categories:

- Regular xfails, such as tests for a new operator (for example `Col2Im`) which we do not support yet. These will be corrected in their own pace.
- More groundbreaking xfails, such as ONNX Serialization. These will be fixed ASAP, but we would rather not mix bugfixing with upgrading the ONNX package.

File: src/frontends/onnx/tests/requirements.txt
Comment (ilyachur) [inline src/frontends/onnx/tests/requirements.txt:3]: Can we update C++ version only for ONNX?

Comment (p-wysocki) [reply src/frontends/onnx/tests/requirements.txt:3]: Done, the CI is running and will show potential issues, if any.

**Review by ilya-lavrenov — APPROVED** [review]

File: thirdparty/onnx/CMakeLists.txt
Comment (ilya-lavrenov) [inline thirdparty/onnx/CMakeLists.txt:22]: why not under MSVC?

Comment (p-wysocki) [reply thirdparty/onnx/CMakeLists.txt:22]: It's just a test for now - it was under MSVC in the previous commit, but ONNX was still being compiled with C++14. I switched to this `if` to see if it works now.

Comment (p-wysocki) [reply thirdparty/onnx/CMakeLists.txt]: I moved the check back under MSVC, it compiles now. The PR is still blocked by a bug with ONNX serialization related to protobuf upgrade.

**Comment by p-wysocki** [comment]
Currently blocked by an ONNX serialization bug related to protobuf upgrade.

**Review by rkazants — CHANGES_REQUESTED** [review]
Now if we update this onnx version, will we get incompatible two environments TF and ONNX, right? How we are going to test MO for which we install all extras and validate models? Will we break testing, DLB, e2e?

Also, please resolve conflicting files.

**Comment by p-wysocki** [comment]
@rkazants Those are all fair points, and the discussion is ongoing in internal channels. Additionally, the PR is also blocked by an ONNX serialization issue and will not be merged anytime soon.

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:331]: @popovaan, can you please take a look? Your test is switched off, how is it critical?
Please look at the code -may be something breaks it.

File: .ci/azure/linux.yml
Comment (rkazants) [inline .ci/azure/linux.yml:390]: wow, not a good think to switch off all PDPD tests.
@ceciliapeng2011, fyi.

File: tools/mo/requirements_tf2.txt
Comment (rkazants) [inline tools/mo/requirements_tf2.txt:2]: let us not use RC in our requirements files. They are in pre-production state.

File: .ci/azure/linux.yml
Comment (p-wysocki) [reply .ci/azure/linux.yml:390]: They have been switched off on purpose, the PR is now a draft. We are waiting for new pdpd release, and since it has been blocking our CI pipelines, I skipped them in order to complete as much work as possible, instead of waiting to be unblocked. Agreed with @mlukasze.

The tests will be switched back on before marking the PR ready for review and before merge.

File: tools/mo/requirements_tf2.txt
Comment (p-wysocki) [reply tools/mo/requirements_tf2.txt:2]: It's a draft. I used rc0 in order to debug the potential issues, because if we wait for new TensorFlow to officially release we may not have enough time to finish the rest of the work. That way I can continue working, using rc0 as a temporary workaround. 

The version will be switched to official release before marking the PR ready for review and before merge.

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py
Comment (p-wysocki) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:331]: The error message which led me to skipping it can be found in CI for commit `507a7548f664d1f29c90c1bbd707bd3fd26856d3` at https://github.com/openvinotoolkit/openvino/runs/11827363222, `openvino-lin` job.

File: .ci/azure/linux.yml
Comment (p-wysocki) [reply .ci/azure/linux.yml:393]: The current decision is to switch off all pdpd tests on master until new version is released. See 95904.

File: tools/mo/requirements_tf2.txt
Comment (p-wysocki) [reply tools/mo/requirements_tf2.txt:2]: The current decision is to switch off all pdpd tests on master until new version is released. See 95904.

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py
Comment (p-wysocki) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_tf.py:331]: Continued in 106247.

**Review by jiwaszki — APPROVED** [review]

File: .ci/azure/linux.yml
Comment (rkazants) [inline .ci/azure/linux.yml:394]: Now we lose PDPD unit-tests. 
@ceciliapeng2011, @wenjiew, are we ready for this?

**Review by rkazants — CHANGES_REQUESTED** [review]

File: tools/mo/requirements_tf.txt
Comment (rkazants) [inline tools/mo/requirements_tf.txt:8]: why do we need this? Please remove it

File: src/frontends/tensorflow/tests/requirements.txt
Comment (p-wysocki) [reply src/frontends/tensorflow/tests/requirements.txt:2]: We have to use release candidates, because TensorFlow 2.11 has protobuf requirements which are incompatible with ONNX, they are older. TensorFlow 2.12, which release candidates are already on PyPI, relaxes the protobuf requirements and allows us to proceed.

The requirement is written in a way which will install full TensorFlow 2.12 release once it's on PyPI. The only other option to go forward with this PR (and by extension Python 3.11) is to wait for the official release. I insist on using release candidates.

Comment (jiwaszki) [reply src/frontends/tensorflow/tests/requirements.txt:2]: @p-wysocki +1 on this. There is no other way to overcome "rc" tag in plain pip. This is the only working solution that will fit our project.

File: tools/mo/requirements_tf.txt
Comment (p-wysocki) [reply tools/mo/requirements_tf.txt:8]: It has been added because one of the tests started failing after upgrading TensorFlow and protobuf. It can be seen in https://dev.azure.com/openvinoci/dldt/_build/results?buildId=508669&view=logs&j=982848f2-7f8d-5e57-730a-8df1e5f991ba&t=cf9ee82e-e056-500f-81f4-f6a3a79d25db&l=817 (`openino-lin` job, commit `a69a21560dfc0a17afdca9041cf01aea34b01a51`).

A solution proposed in https://stackoverflow.com/questions/75583410/module-save-error-typeerror-this-dict-descriptor-does-not-support-dict recommended specifying exact wrapt version. Doing that resolved the issue.

File: .ci/azure/linux.yml
Comment (ilya-lavrenov) [reply .ci/azure/linux.yml:394]: we have no choice

File: src/frontends/tensorflow/tests/requirements.txt
Comment (rkazants) [reply src/frontends/tensorflow/tests/requirements.txt:2]: It looks that you did these changes to get tests passed and due to our limitations in the validation. Because our validation requires to have installed dependencies for all FWs for conversion of different FW models. So they become incompatible and cannot exist together.

May be we should solve this problem on the validation part and not to change production `requirements` files with RC dependencies. Also, now our customer can potentially install RC on his machine. That is not good!

File: .ci/azure/linux.yml
Comment (rkazants) [reply .ci/azure/linux.yml:394]: Why is it so urgent to update ONNX and Protobuf for the sake of loss of unit-tests?

Comment (ilya-lavrenov) [reply .ci/azure/linux.yml:394]: python 3.11 support

Comment (mlukasze) [reply .ci/azure/linux.yml:394]: It's has been already discussed on various of meeting. Please focus on TF related changes.

File: .ci/azure/linux.yml
Comment (mlukasze) [reply .ci/azure/linux.yml]: @eaidova it's an exact plan of next step, but in same time Roman who liked your proposal has doubts to use not released version for TF, so it need additional discussion.

File: tools/mo/requirements_tf.txt
Comment (rkazants) [reply tools/mo/requirements_tf.txt:8]: @p-wysocki, please put this relaxed dependency into https://github.com/openvinotoolkit/openvino/blob/master/tools/mo/requirements_dev.txt

File: src/frontends/tensorflow/tests/requirements.txt
Comment (mlukasze) [reply src/frontends/tensorflow/tests/requirements.txt:2]: to resolve a discussion - TF 2.12 which satisfies our requirements has been released and this notation is not needed anymore and will be replaced with released-based one.

File: tools/mo/requirements_tf2.txt
Comment (mlukasze) [reply tools/mo/requirements_tf2.txt]: to resolve a discussion - TF 2.12 which satisfies our requirements has been released and this notation is not needed anymore and will be replaced with released-based one.

File: tools/mo/requirements_tf.txt
Comment (p-wysocki) [reply tools/mo/requirements_tf.txt]: I skipped the test for now and created 106651. I hope that's okay with you.

**Review by rkazants — APPROVED** [review]

**Pull request merged**
